pennylane/queuing.py

```diff
@@ -11,8 +11,106 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""
-This module contains the :class:`QueuingManager`.
+r"""
+This module contains the classes for placing objects into queues.
+
+Description
+-----------
+
+A *queuable object* is anything that can be placed into a queue. These will be :class:`~.Operator`,
+:class:`~.MeasurementProcess`, and :class:`~.QuantumTape` objects. :class:`~.Operator` and
+:class:`~.MeasurementProcess` objects achieve queuing via a :meth:`~.Operator.queue` method called upon construction.
+Note that even though :class:`~.QuantumTape` is a queuable object, it does not have a ``queue`` method.
+
+When an object is queued, it sends itself to the :class:`~.QueuingManager`. The :class:`~.QueuingManager`
```

Comment:
internally it's basically just a last-in, first-out (LIFO) queue of active contexts. Whatever context (aka tape aka AnnotatedQueue) you're in will be the last-in to this list, so any operators being queued will be pushed onto that list. A bit funny to think about because the QueuingManager watches over a queue of queues!

Reply:
Added an example of nested queues.  

But basically `QueuingManager` is global, so globally only one queue is recording at any point.